canvas/fizz-27-py/src/data-extractor.py

```diff
@@ -0,0 +1,40 @@
+from lib import exportData
+from dotenv import load_dotenv
+from lib.courses import getCoursesForYear
+from lib.students import getStudentsForCourse, getEnrollmentsForStudent
+from lib.assignments import getAssignmentsForCourse
+from lib.submissions import getSubmissionsForAssignment
+
+load_dotenv()
+
+all_courses = getCoursesForYear()
+all_students = list()
+all_assignments = list()
+all_submissions = list()
+all_enrollments = list()
+
+for course in all_courses:
+    course_id = course["id"]
+    students = getStudentsForCourse(course_id)
+    assignments = getAssignmentsForCourse(course_id)
+
+    for assignment in assignments:
+        assignment_id = assignment["id"]
+        submissions_for_assignment = getSubmissionsForAssignment(course_id, assignment_id)
+        all_submissions = all_submissions + submissions_for_assignment
+
+    students = list(filter(lambda student: student not in all_students, students))
+    all_students = all_students + students
```

Comment:
If `all_students` is intended not to have duplicates, then it may make sense to create a `set` instead of a `list`. This then simplifies lines 26 and 27 to `all_students = all_students | set(students)`.